climpred/tests/test_metrics.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+import pytest
+import xarray as xr
+import xskillscore as xs
+from climpred.constants import PM_COMPARISONS
+from climpred.prediction import compute_hindcast, compute_perfect_model
+from climpred.tutorial import load_dataset
+from xarray.testing import assert_allclose
+
+
+@pytest.fixture
+def pm_da_ds1d():
+    da = load_dataset('MPI-PM-DP-1D')
+    da = da['tos'].isel(area=1, period=-1)
+    return da
+
+
+@pytest.fixture
+def pm_da_control1d():
+    da = load_dataset('MPI-control-1D')
+    da = da['tos'].isel(area=1, period=-1)
+    return da
+
+
+@pytest.fixture
+def ds_3d_NA():
+    """ds North Atlantic"""
+    ds = load_dataset('MPI-PM-DP-3D')['tos'].sel(x=slice(120, 130), y=slice(50, 60))
+    return ds
+
+
+@pytest.fixture
+def control_3d_NA():
+    """control North Atlantic"""
+    ds = load_dataset('MPI-control-3D')['tos'].sel(x=slice(120, 130), y=slice(50, 60))
+    return ds
+
+
+@pytest.fixture
+def initialized_da():
+    da = load_dataset('CESM-DP-SST-3D')['SST']
+    da = da - da.mean('init')
+    return da
+
+
+@pytest.fixture
+def reconstruction_da():
+    da = load_dataset('FOSI-SST-3D')['SST']
+    da = da - da.mean('time')
+    return da
+
+
+def my_mse(forecast, reference, dim='svd', **metric_kwargs):
+    return ((forecast - reference) ** 2).mean(dim)
+
+
+@pytest.mark.skip(reason='not implemented yet')
+@pytest.mark.parametrize('comparison', PM_COMPARISONS)
+@pytest.mark.parametrize('metric', [my_mse])
+def test_new_metric_passed_to_compute(pm_da_ds1d, pm_da_control1d, metric, comparison):
+    actual = compute_perfect_model(
+        pm_da_ds1d, pm_da_control1d, comparison=comparison, metric=metric
+    )
+
+    expected = compute_perfect_model(
+        pm_da_ds1d, pm_da_control1d, comparison=comparison, metric='mse'
+    )
+
+    assert_allclose(actual, expected)
+
+
+@pytest.mark.parametrize('metric', ('rmse', 'mse'))
+def test_pm_metric_skipna(ds_3d_NA, control_3d_NA, metric):
+    ds_3d_NA = ds_3d_NA.copy()
+    # manipulating data
+    # problem here: i dont get a nice example
+    ds_3d_NA.data[0, 0, 0, 80:100, 80:100] = np.nan
```

Comment:
I dont a suitable example to test `skipna`. any ideas? I want to mask an area in one member and compare compute(skipna=False) with compute(skipna=True)